pkgs/applications/misc/jetbrains-toolbox/default.nix

```diff
@@ -0,0 +1,119 @@
+{ stdenv
+, lib
+, glibc
+, xcbutilkeysyms
+, gcc
+, zlib
+, libsecret
+, xorg
+, libcef
+, libdbusmenu-gtk3
+, libdbusmenu
+, pcre
+, jetbrains
+, fetchzip
+, copyDesktopItems
+, makeDesktopItem
+, makeWrapper
+, runCommand
+, appimageTools
+, patchelf
+}:
+let
+  pname = "jetbrains-toolbox";
+  version = "1.27.3.14493";
+
+  src = fetchzip {
+    url = "https://download.jetbrains.com/toolbox/jetbrains-toolbox-${version}.tar.gz";
+    sha256 = "sha256-aK5T95Yg8Us8vkznWlDHnPiPAKiUtlU0Eswl9rD01VY=";
+    stripRoot = false;
+  };
+
+  libPath = lib.makeLibraryPath [
+    stdenv.cc.cc.lib
+    glibc
+    xorg.xcbutilkeysyms
+    gcc
+    zlib
+    libsecret
+    xorg.libXext
+  ];
+
+  extractApp = { name, src }: runCommand "${name}-extracted"
+    {
+      buildInputs = [ appimageTools.appimage-exec patchelf gcc libcef ];
+    }
+    ''
+    appimage-exec.sh -x $out ${src}/${name}-${version}/${name}
+    patchelf \
+      --set-rpath "${libPath}:$out/jre/lib:/usr/lib64" \
+      --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" \
+      $out/jetbrains-toolbox
+    patchelf --set-interpreter "${stdenv.cc.bintools.dynamicLinker}" $out/glibcversion
+  '';
+
+  appimageContents = extractApp {
+    inherit src;
+    name = pname;
+  };
+
+  appimage = appimageTools.wrapAppImage {
+    inherit pname version;
+    src = appimageContents;
+    extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.targetPkgs pkgs)
+      ++ (with pkgs; [
+      libsecret
+      libcef
+      zlib
+      xorg.xcbutilkeysyms
+      libdbusmenu-gtk3
+      libdbusmenu
+      pcre
+      jetbrains.jdk
+      xorg.xcbutilkeysyms
+      xorg.libXext
+    ]);
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "JetBrains Toolbox";
+    exec = "jetbrains-toolbox";
+    comment = "JetBrains Toolbox";
+    desktopName = "JetBrains Toolbox";
+    type = "Application";
+    icon = "/run/current-system/sw/share/icons/jetbrains-toolbox.svg";
+    terminal = false;
+    categories = [ "Development" ];
+    startupWMClass = "jetbrains-toolbox";
+    startupNotify = false;
+  };
+
+in
+stdenv.mkDerivation {
+  inherit pname version src appimage;
+
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -pv $out/share/applications
+    mkdir -pv $out/share/icons
+    cp -r ${appimageContents}/.DirIcon $out/share/icons/jetbrains-toolbox.svg
```

Comment:
I'm not sure about those 3 first lines, shouldn't this be handled in `desktopItem` ?